### DISCUSSION TITLE
Allow CircleCI caches to be manually cleared

### DISCUSSION THREAD

**Discussion by gsmendoza** [solidusio/solidus#4759] [New Features or Ideas]
## Goal

I would like to insert a UI environment variable to our CircleCI cache keys so that we can manually clear them if needed.

## Background

We recently have a situation where the Ruby version in the [`install_solidus` cache key](https://github.com/solidusio/solidus/blob/1829fc87e769acd8023c6321d593861735e7cb18/.circleci/config.yml#L179)  doesn't match the Ruby version of the Rails app that was cached. As a result, the mismatch is causing the following error in CI:

```
Your Ruby version is 2.7.7, but your Gemfile specified 2.7.6
```

For examples, see:

* https://app.circleci.com/pipelines/github/solidusio/solidus/3838/workflows/2b5f5731-1c8c-4589-bfc0-6206156cd364/jobs/34317?invite=true#step-108-130
* https://app.circleci.com/pipelines/github/nebulab/solidus/625/workflows/effbd9a1-984d-44ca-bfc1-681de35647ed/jobs/7598?invite=true#step-108-128

## Proposed implementation

Basing from https://support.circleci.com/hc/en-us/articles/115015426888-Clear-Project-Dependency-Cache, we can insert a UI environment variable to our two cache keys in our CircleCI config:

* `solidus-gems-v3-{{ .Environment.CACHE_VERSION }}-{{checksum ".ruby-version"}}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}`
* `solidus-installer-v5-{{ .Environment.CACHE_VERSION }}-{{ checksum "/tmp/.tool-versions" }}`

We'll keep the existing cache version keys (v3 and v5) in the cache keys so that we'll still have the ability to reset the cache via source code change.

## Prerequisites

To be safe, we need to seed the `CACHE_VERSION` environment variable in CircleCI before merging the PR to master.

## Post-merge changes

Once merged to master, we'll need to backport the change to older versions of Solidus.

## Additional references

* https://circleci.com/docs/set-environment-variable/#set-an-environment-variable-in-a-project

**Comment by waiting-for-dev** [comment]
I like the idea, @gsmendoza!

> Once merged to master, we'll need to
>
>  - Seed the CACHE_VERSION environment variable in CircleCI.

We probably need to do that BEFORE merging to master, to avoid the environment variable not being found.

**Reply by gsmendoza** [reply]
Yep, I agree. I updated the description above :+1:
